processor/routingprocessor/config.go

```diff
@@ -64,23 +65,27 @@ type Config struct {
 
 // Validate checks if the processor configuration is valid.
 func (c *Config) Validate() error {
+	// validate that there's at least one item in the table
+	if len(c.Table) == 0 {
+		return fmt.Errorf("invalid routing table: %w", errNoTableItems)
+	}
+
 	// validate that every route has a value for the routing attribute and has
 	// at least one exporter
 	for _, item := range c.Table {
-		if len(item.Value) == 0 {
+		if len(item.Value) == 0 && len(item.Expression) == 0 {
 			return fmt.Errorf("invalid (empty) route : %w", errEmptyRoute)
 		}
 
+		if len(item.Value) != 0 && len(item.Expression) != 0 {
+			return fmt.Errorf("both expression and value provided")
```

Comment:
We can provide more info as part of this error message. Something like:

```suggestion
			return fmt.Errorf("invalid route: both expression (%q) and value (%q) provided", item.Expression, item.Value)
```